model/word2vec/state.go

```diff
@@ -95,49 +103,55 @@ func (s *State) Preprocess(f io.ReadSeeker) (io.ReadCloser, error) {
 }
 
 // Trainer trains a corpus. It assumes that Preprocess() has already been called
-func (s *State) Trainer(f io.ReadCloser, trainOne func(wordIDs []int, wordIndex int) error) error {
+func (s *State) Trainer(f io.ReadCloser, trainOne func(wordIDs []int, wordIndex int, lr float64) error) error {
 	s.startTraining()
 	defer func() {
 		s.endTraining()
 		f.Close()
 	}()
 
+	go s.incrementDoneWord()
+
+	errChan := make(chan error, 1)
+	sema := make(chan struct{}, runtime.NumCPU())
+	var wg sync.WaitGroup
+
 	var current int
```

Comment:
[nits] I confused this `current` with `currentWordSize` (it's not in this PR target, but I'd like to change this name). What do you think about?

For example:
- current: buffer (chunk)
- currentWordSize: stay this name or `trainedWordSize`

Reply:
I agree with both changes. I'll put both in

Reply:
Changes:

`current`: `buffered`
`currentWordSize`: `trainedWords`
`ignoreWord`: `ignoredWords`